internal/provider/be_api_key_resource.go

```diff
@@ -0,0 +1,222 @@
+		)
+		return
+	}
+
+	// set the Terraform state.
+	plan.Name = types.StringValue(beApiKeyInfo.Name)
+	plan.ApiKey = types.StringValue(beApiKeyInfo.ApiKey)
+	plan.ApiKeyId = types.StringValue(beApiKeyInfo.ApiKeyId)
+	plan.ReadOnly = types.BoolValue(beApiKeyInfo.IsReadOnly)
+
+
+	// Write logs using the tflog package
+	// Documentation: https://terraform.io/plugin/log
+	tflog.Trace(ctx, "created a propelauth_be_api_key resource")
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+func (r *beApiKeyResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	// Get current state and read it into the model
+	var state beApiKeyResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// retrieve the be api key from PropelAuth
+	beApiKeyInfo, err := r.client.GetBeApiKeyInfo(state.Environment.ValueString(), state.ApiKeyId.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error Reading PropelAuth Backend API Key",
+			"Could not read PropelAuth Backend API Key: "+err.Error(),
+		)
+		return
+	}
+
+	// update the state for the be api key
+	state.Name = types.StringValue(beApiKeyInfo.Name)
+	state.ApiKeyId = types.StringValue(beApiKeyInfo.ApiKeyId)
+	state.ReadOnly = types.BoolValue(beApiKeyInfo.IsReadOnly)
+	// state.ApiKey is omitted, as it is stripped due to containing sensitive data.
+	// When created, it contains the full API Key, but when read, it is stripped.
+	// Thus, it is not necessary to update the state with the API Key.
+
+	// Save updated state into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (r *beApiKeyResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// Read Terraform plan data into the model
+	var plan beApiKeyResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Update the be api key
+	beApiKeyResponse, err := r.client.UpdateBeApiKey(plan.Environment.ValueString(), plan.ApiKeyId.ValueString(), plan.Name.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error updating be api key",
+			"Could not update the be api key, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	// Save updated state from the response into Terraform state
+	plan.Name = types.StringValue(beApiKeyResponse.Name)
+	plan.ReadOnly = types.BoolValue(beApiKeyResponse.IsReadOnly)
+
+	tflog.Trace(ctx, "updated a propelauth_be_api_key resource")
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+func (r *beApiKeyResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state beApiKeyResourceModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Delete existing order
```

Comment:
tiny bit of comment clean up: I'm guessing "order" was just from copy-pasting template code